### PR TITLE
feat: Add create issue credential

### LINKED ISSUE #2: Create Credential Issuer Rest API

### PR DESCRIPTION
**Pull request by talwinder50**
closes #2

Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

### CODE REVIEW

File: pkg/restapi/issuer/operation/operations.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	issueDate := time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(),
+		time.Now().Hour(), time.Now().Minute(), time.Now().Second(), 0, time.UTC)
+
+	// todo Issue raised, only base context doesnt work
+	credential.Context = []string{credentialContext, data.Context}
+	credential.Subject = data.Subject
+	credential.Types = data.Type
+	credential.Issuer = data.Issuer
+	credential.Issued = &issueDate
+	credential.Schemas = []verifiable.TypedID{}
```
Comment (talwinder50) [inline pkg/restapi/issuer/operation/operations.go]: Gonna remove this , as the bug was raised to dima and he fixed

File: pkg/restapi/issuer/operation/operations.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	}
+}
+
+func createCredential(data *Issuer) (*verifiable.Credential, error) {
+	credential := &verifiable.Credential{}
+	issueDate := time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(),
+		time.Now().Hour(), time.Now().Minute(), time.Now().Second(), 0, time.UTC)
+
+	// todo Issue raised, only base context doesnt work
+	credential.Context = []string{credentialContext, data.Context}
```
Comment (talwinder50) [inline pkg/restapi/issuer/operation/operations.go]: Gonna remove this , as the bug was raised to dima and he fixed

File: cmd/issuer-rest/startcmd/start.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+const (
+	hostURLFlagName      = "host-url"
+	hostURLFlagShorthand = "u"
+	hostURLFlagUsage     = "URL to run the issuer-rest instance on. Format: HostName:Port."
+	hostURLEnvKey        = "ISSUER_REST_HOST_URL"
+)
+
+var errMissingHostURL = errors.New("host URL not provided")
+
+type edgeStoreParameters struct {
```
Comment (DRK3) [inline cmd/issuer-rest/startcmd/start.go]: Change name to issuerRestParameters or something similar

File: pkg/restapi/issuer/operation/operations.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+	credential.Context = []string{credentialContext}
+	credential.Subject = data.Subject
+	credential.Types = data.Type
+	credential.Issuer = data.Issuer
+	credential.Issued = &issueDate
+	credential.ID = ID
+
+	cred, err := json.Marshal(credential)
+
```
Comment (DRK3) [inline pkg/restapi/issuer/operation/operations.go]: Remove the line of whitespace here (line 94) for consistency

Comment (talwinder50) [reply pkg/restapi/issuer/operation/operations.go]: done

File: cmd/issuer-rest/startcmd/start.go (hunk 3 above)
Comment (talwinder50) [reply cmd/issuer-rest/startcmd/start.go]: done

File: pkg/restapi/issuer/operation/operations_test.go [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+			return h
+		}
+	}
+
+	require.Fail(t, "unable to find handler")
+
+	return nil
+}
+
+type mockResponseWriter struct {
```
Comment (DRK3) [inline pkg/restapi/issuer/operation/operations_test.go]: Small detail, but I would name this more specifically so we know it always fails

Comment (talwinder50) [reply pkg/restapi/issuer/operation/operations_test.go]: I could not think of other name .. do you have any suggestion

Comment (DRK3) [reply pkg/restapi/issuer/operation/operations_test.go]: failingResponseWriter?

Comment (DRK3) [reply pkg/restapi/issuer/operation/operations_test.go]: If you don't think that's an improvement though, then I'm ok with mockResponseWriter

File: pkg/restapi/vc/operation/operations.go [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+		if err != nil {
+			log.Errorf("Failed to write response for credential creation failure (unable to read request): %s", err.Error())
+		}
+	}
+
+	cred, err := createCredential(&data)
+
+	if err != nil {
+		log.Errorf("Failed to create new credential: %s", err.Error())
```
Comment (sandrask) [inline pkg/restapi/vc/operation/operations.go]: you need error status and return here

File: pkg/restapi/vc/operation/operations.go [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	rw.WriteHeader(http.StatusCreated)
+
+	resp, err := json.Marshal(cred)
+	if err != nil {
+		log.Errorf("Failed to marshal credential: %s", err.Error())
```
Comment (sandrask) [inline pkg/restapi/vc/operation/operations.go]: same as above: set error status and return here

File: pkg/restapi/vc/operation/operations.go [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	_, err = rw.Write(resp)
+	if err != nil {
+		log.Errorf("Failed to write response for VCInputData vault creation success: %s", err.Error())
```
Comment (sandrask) [inline pkg/restapi/vc/operation/operations.go]: same as above: set error status and return here

Comment (DRK3) [reply pkg/restapi/vc/operation/operations.go]: Good catch - I missed this earlier when reviewing

File: pkg/restapi/vc/operation/operations_test.go [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+	createCredentialHandler := getHandler(t, op, createCredentialEndpoint)
+
+	req, err := http.NewRequest(http.MethodPost, "/credentials", bytes.NewBuffer([]byte("")))
+	require.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+
+	createCredentialHandler.Handle().ServeHTTP(rr, req)
+
+	require.Equal(t, http.StatusBadRequest, rr.Code)
```
Comment (DRK3) [inline pkg/restapi/vc/operation/operations_test.go]: Something else to check out...it looks like your handler will always return StatusCreated, so I'm curious why this test actually passes. (The subsequent call to WriteHeader is ignored?)

File: pkg/restapi/vc/operation/operations.go (hunk 6 above)
Comment (talwinder50) [reply pkg/restapi/vc/operation/operations.go]: done

File: pkg/restapi/vc/operation/operations.go (hunk 7 above)
Comment (talwinder50) [reply pkg/restapi/vc/operation/operations.go]: done

File: pkg/restapi/vc/operation/operations.go (hunk 8 above)
Comment (talwinder50) [reply pkg/restapi/vc/operation/operations.go]: done

File: pkg/restapi/vc/operation/operations_test.go (hunk 9 above)
Comment (talwinder50) [reply pkg/restapi/vc/operation/operations_test.go]: updated

File: pkg/restapi/vc/operation/operations.go (hunk 6 above)
Comment (DRK3) [reply pkg/restapi/vc/operation/operations.go]: Should InternalServerError be used instead of StatusNotAcceptable? @sandrask what do you think?

Comment (talwinder50) [reply pkg/restapi/vc/operation/operations.go]: Chaged this to StatusBadRequest

**Pull request merged**